Sources/Common/Core/Math/index.js

```diff
@@ -530,44 +532,29 @@ export function linearSolve3x3(mat_3x3, x_3, y_3) {
 }
 
 export function multiply3x3_vect3(mat_3x3, in_3, out_3) {
-  const x =
-    mat_3x3[0][0] * in_3[0] + mat_3x3[0][1] * in_3[1] + mat_3x3[0][2] * in_3[2];
-  const y =
-    mat_3x3[1][0] * in_3[0] + mat_3x3[1][1] * in_3[1] + mat_3x3[1][2] * in_3[2];
-  const z =
-    mat_3x3[2][0] * in_3[0] + mat_3x3[2][1] * in_3[1] + mat_3x3[2][2] * in_3[2];
+  const x = mat_3x3[0] * in_3[0] + mat_3x3[1] * in_3[1] + mat_3x3[2] * in_3[2];
+  const y = mat_3x3[3] * in_3[0] + mat_3x3[4] * in_3[1] + mat_3x3[5] * in_3[2];
+  const z = mat_3x3[6] * in_3[0] + mat_3x3[7] * in_3[1] + mat_3x3[8] * in_3[2];
 
   out_3[0] = x;
   out_3[1] = y;
   out_3[2] = z;
 }
 
 export function multiply3x3_mat3(a_3x3, b_3x3, out_3x3) {
-  const tmp = [
-    [0, 0, 0],
-    [0, 0, 0],
-    [0, 0, 0],
-  ];
-
-  for (let i = 0; i < 3; i++) {
-    tmp[0][i] =
-      a_3x3[0][0] * b_3x3[0][i] +
-      a_3x3[0][1] * b_3x3[1][i] +
-      a_3x3[0][2] * b_3x3[2][i];
-    tmp[1][i] =
-      a_3x3[1][0] * b_3x3[0][i] +
-      a_3x3[1][1] * b_3x3[1][i] +
-      a_3x3[1][2] * b_3x3[2][i];
-    tmp[2][i] =
-      a_3x3[2][0] * b_3x3[0][i] +
-      a_3x3[2][1] * b_3x3[1][i] +
-      a_3x3[2][2] * b_3x3[2][i];
+  const copyA = createArray(9);
+  const copyB = createArray(9);
+  for (let j = 0; j < 9; j++) {
+    copyA[j] = a_3x3[j];
```

Comment:
it is useless to `createArray` filled with 0s (more costly than `new Array(9)` ) and overwrite it just after

Reply:
use `a_3x3.slice(0, 9)`